test/integration/bind_integration_test.cc

```diff
@@ -0,0 +1,121 @@
+#include <string>
+
+#include "common/http/header_map_impl.h"
+#include "common/protobuf/utility.h"
+
+#include "test/integration/integration.h"
+#include "test/integration/utility.h"
+#include "test/test_common/network_utility.h"
+#include "test/test_common/utility.h"
+
+#include "gtest/gtest.h"
+
+namespace Envoy {
+namespace {
+
+class BindIntegrationTest : public BaseIntegrationTest,
```

Comment:
Preference for leaving the test in the same file, with a TODO for me to pull out all the annoying config gorp into a helper class in a follow-up

If you'd prefer having a separate file let's make it generic YamlIntegrationTest or ProtoIntegrationTest where we encourage others adding new features to drop them here, with a TODO for me to pull the bind-specific logic into just the Bind test.

Reply:
I will make this `ProtoIntegrationTest` and add a TODO for you.